src/main/java/ua/lviv/iot/parkingServer/natscontroller/NatsParkingGetByIdController.kt

```diff
@@ -20,15 +21,15 @@ class NatsParkingGetByIdController(
 
     override val parser: Parser<GetByIdParkingRequest> = GetByIdParkingRequest.parser()
 
-    override fun generateReplyForNatsRequest(
-        request:GetByIdParkingRequest
-    ): GetByIdParkingResponse {
-        val parkingById = service.findEntityById(request.parkingId)
-        val protoParking = converter.parkingToProto(parkingById)
+    override fun generateReplyForNatsRequest(request: GetByIdParkingRequest): Mono<GetByIdParkingResponse> {
+        val parkingId = request.parkingId
 
-        return GetByIdParkingResponse
-            .newBuilder()
-            .setParking(protoParking)
-            .build()
+        return service.findEntityById(parkingId)
+            .map { parking ->
+                val protoParking = converter.parkingToProto(parking)
+                GetByIdParkingResponse.newBuilder()
+                    .setParking(protoParking)
+                    .build()
+            }
     }
```

Comment:
```suggestion
        return service.findEntityById(parkingId)
            .map { converter.parkingToProto(it) }
            .map {
                GetByIdParkingResponse.newBuilder()
                    .setParking(it)
                    .build()
            }
    }
```

you have two mapping operations and it's better to split them into two reactor steps. also, do not afraid to use it

Reply:
✅